diff --git a/src/ui/app-grid.ui b/src/ui/app-grid.ui
index 0bf0cc1..4077033 100644
--- a/src/ui/app-grid.ui
+++ b/src/ui/app-grid.ui
@@ -32,6 +32,7 @@
                             <property name="halign">center</property>
                             <property name="hexpand">1</property>
                             <property name="has-frame">0</property>
+                            <property name="input-purpose">free-form</property>
                             <property name="max-width-chars">48</property>
                             <signal name="search-changed" handler="on_search_changed" object="PhoshAppGrid" swapped="no"/>
                             <signal name="preedit-changed" handler="on_search_preedit_changed" object="PhoshAppGrid" swapped="no"/>
diff --git a/src/ui/top-panel.ui b/src/ui/top-panel.ui
index 1dd36b1..a49756f 100644
--- a/src/ui/top-panel.ui
+++ b/src/ui/top-panel.ui
@@ -272,25 +272,30 @@
                         </style>
                         <child>
-                          <object class="PhoshStatusIconsBox" id="status_icons_box">
+                          <object class="GtkBox">
                             <property name="visible">1</property>
                             <property name="spacing">8</property>
-                            <property name="align">end</property>
-                            <child>
-                              <object class="GtkLabel" id="lbl_lang">
-                                <property name="visible">0</property>
-                                <property name="margin-top">2</property>
-                                <property name="justify">center</property>
-                              </object>
-                            </child>
-                            <child><object class="PhoshLocationInfo" id="top_location"><property name="visible">1</property><property name="pixel-size">13</property></object></child>
-                            <child><object class="GtkImage"><property name="visible">1</property><property name="icon-name">security-high-symbolic</property><property name="pixel-size">13</property></object></child>
+                            <child><object class="PhoshWifiInfo" id="top_wifi"><property name="visible">1</property><property name="pixel-size">13</property></object></child>
                             <child>
                               <object class="PhoshBatteryInfo" id="batteryinfo">
                                 <property name="visible">1</property>
                                 <property name="pixel-size">13</property>
                               </object>
                             </child>
-                            <child><object class="PhoshWifiInfo" id="top_wifi"><property name="visible">1</property><property name="pixel-size">13</property></object></child>
+                            <child><object class="GtkImage"><property name="visible">1</property><property name="icon-name">security-high-symbolic</property><property name="pixel-size">13</property></object></child>
+                            <child><object class="PhoshLocationInfo" id="top_location"><property name="visible">1</property><property name="pixel-size">13</property></object></child>
+                          </object>
+                        </child>
+                        <child>
+                          <object class="PhoshStatusIconsBox" id="status_icons_box">
+                            <property name="visible">0</property>
+                            <property name="no-show-all">1</property>
+                            <child>
+                              <object class="GtkLabel" id="lbl_lang">
+                                <property name="visible">0</property>
+                                <property name="margin-top">2</property>
+                                <property name="justify">center</property>
+                              </object>
+                            </child>
                           </object>
                         </child>
                       </object>
